pkg/controller/daemon/update.go

```diff
@@ -162,6 +176,15 @@ func (dsc *DaemonSetsController) rollingUpdate(ctx context.Context, ds *apps.Dae
 			case !podutil.IsPodAvailable(oldPod, ds.Spec.MinReadySeconds, metav1.Time{Time: now}):
 				// the old pod isn't available, allow it to become a replacement
 				logger.V(5).Info("Pod on node is out of date and not available, allowing replacement", "daemonset", klog.KObj(ds), "pod", klog.KObj(oldPod), "node", klog.KRef("", nodeName))
+
+				node, err := dsc.nodeLister.Get(nodeName)
+				if err != nil {
+					return fmt.Errorf("couldn't get node for nodeName %q: %v", nodeName, err)
```

Comment:
@atiratree 
[The same reason](https://github.com/kubernetes/kubernetes/pull/119317/files#r1316715032) when get pod, I've changed to emitting log and skip the node instead of `fmt.Errorf`.

Reply:
I think it is preferable to keep the errors: please see https://github.com/kubernetes/kubernetes/pull/119317#discussion_r1320206492